event/src/main/java/com/bazaarvoice/emodb/event/db/astyanax/DefaultSlabAllocator.java

```diff
@@ -94,7 +96,7 @@ public SlabAllocation allocate(String channelName, int desiredCount) {
             // SlabPersister I/O operation around creating slabs.
 
             // Is there is an existing open slab we can allocate from?
-            SlabAllocation allocation = channelState.allocate(desiredCount);
+            SlabAllocation allocation = channelState.allocate(eventSizes);
             if (allocation != null) {
                 return allocation;
             }
```

Comment:
github won't let me comment on the few lines after this, but we can't allocate a slab with `Constants.MAX_SLAB_SIZE` because we dont actually know if we can fit that events in the slab. If we want to keep this code, we still need to run some sort of method that drains eventSizes, even if we don't ever attach the slab. Additionally, this current functionality does drain eventSizes, event though it gives a slab of size 1000

Reply:
OK.